File: Data Structures & Algorithms/three-integer-sum/submission-4.cpp
class Solution {
public:
    vector<vector<int>> threeSum(vector<int>& nums) {
        vector<vector<int>> res;
        sort(nums.begin(), nums.end());
        for (int i = 0; i < nums.size(); i++){
            if (i > 0 && nums[i] == nums[i - 1]){
                continue;
            }

            int l = i + 1;
            int r = nums.size() - 1;
            while (l < r){
                int threeSums = nums[i] + nums[l] + nums[r];
                if (threeSums > 0){
                    r--;
                }
                else if (threeSums < 0) {
                    l++;
                }
                else{
                    res.push_back({nums[i], nums[l], nums[r]});
                    l++;
                    while (nums[l] == nums[l - 1] && l < r){
                        l++;
                    }
                }
            }
        }
        return res;
    }
};

File: Data Structures & Algorithms/top-k-elements-in-list/submission-4.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        counter_map = {}
        for num in nums:
            counter_map[num] = counter_map.get(num,0) + 1
        print(counter_map)
        res = []

        _map = [[] for _ in range((len(nums) + 1))]
        print(_map)
        for key,value in counter_map.items():
            _map[value].append(key)
        
        print(_map)
        for i in range(len(_map) - 1, 0, -1):
            for num in _map[i]:
                if k > 0:
                    res.append(num)
                    k -= 1
                if k == 0:
                    return res
        print(res) 
        return [] 
        
